src/shiver/views/histogram_parameters.py

```diff
@@ -0,0 +1,532 @@
+        if state == QtGui.QValidator.Acceptable:
+            color = "#ffffff"
+        elif state == QtGui.QValidator.Intermediate:
+            color = "#ffaaaa"
+        else:
+            color = "#ff0000"
+        sender.setStyleSheet("QLineEdit { background-color: %s }" % color)
+        if state == QtGui.QValidator.Acceptable:
+            # if value is acceptable check all three projections
+            self.validateprojection_values()
+
+    def validateprojection_values(self):
+        """Validate the projection values on co-linear"""
+        color = "#ff0000"
+        # examine whether the projections are all non-colinear
+        if (
+            self.projection_u.validator().validate(self.projection_u.text(), 0)[0] == QtGui.QValidator.Acceptable
+            and self.projection_v.validator().validate(self.projection_v.text(), 0)[0] == QtGui.QValidator.Acceptable
+            and self.projection_w.validator().validate(self.projection_w.text(), 0)[0] == QtGui.QValidator.Acceptable
+        ):
+            b1 = numpy.fromstring(str(self.projection_u.text()), sep=",")
+            b2 = numpy.fromstring(str(self.projection_v.text()), sep=",")
+            b3 = numpy.fromstring(str(self.projection_w.text()), sep=",")
+            if numpy.abs(numpy.inner(b1, numpy.cross(b2, b3))) > 1e-5:
+                color = "#ffffff"
+                self.projections_valid_state = True
+                self.dimensions.update_dimension_names([b1, b2, b3])
+        self.projection_u.setStyleSheet("QLineEdit { background-color: %s }" % color)
+        self.projection_v.setStyleSheet("QLineEdit { background-color: %s }" % color)
+        self.projection_w.setStyleSheet("QLineEdit { background-color: %s }" % color)
+
+    def set_dimension(self, btn):
+        """Based on the radio button step, allow the corresponding step values to be filled in;
+        the rest become read-only"""
+
+        # initialize colors and readOnly fields
+        color1 = color2 = color3 = color4 = "#ffffff"
+
+        if self.dimensions.combo_step1.text() == "":
+            color1 = "#ffaaaa"
+        if self.dimensions.combo_step2.text() == "":
+            color2 = "#ffaaaa"
+            self.dimensions.combo_step2.setReadOnly(False)
+        if self.dimensions.combo_step3.text() == "":
+            color3 = "#ffaaaa"
+            self.dimensions.combo_step3.setReadOnly(False)
+        if self.dimensions.combo_step4.text() == "":
+            color4 = "#ffaaaa"
+            self.dimensions.combo_step4.setReadOnly(False)
+
+        self.required_steps = [self.dimensions.combo_step1]
+        if btn.isChecked():
+            # if text exists in the steps that cannot be filled in,
+            # it is cleared and banckground color is changed
+            if btn.text() == self.btn_dimensions[0]:
+                color2 = "#ffffff"
+                self.dimensions.combo_step2.setText("")
+                self.dimensions.combo_step2.setReadOnly(True)
```

Comment:
Instead of making it read only, how about making it invisible `.setVisible(Faslse)`